src/jit/gentree.h

```diff
@@ -4112,6 +4112,7 @@ struct GenTreeSIMD : public GenTreeJitIntrinsic
 struct GenTreeHWIntrinsic : public GenTreeJitIntrinsic
 {
     NamedIntrinsic gtHWIntrinsicId;
+    var_types      gtIndexBaseType; // for AVX2 Gather* intrinsics
```

Comment:
It is probably not a big issue for intrinsics, though for methods with heavy intrinsic usage it could be an impact. Did you consider deriving from `GenTreeHWIntrinsic` to make a specialized node?

Reply:
It might also be worth considering whether to define something like `GenTreeHWIntrinsicBig`, to isolate both the additional fields as well as the extra operands.

Reply:
> Perhaps make NamedIntrinsic unsigned short? 64k intrinsics ought to be enough for anybody.

@mikedn Good point! Yes, 64k is definitely enough for the foreseeable future (AVX-512 based ISAs).

Reply:
> make NamedIntrinsic unsigned short

Made this change, now `GenTreeHWIntrinsic` is still a small node.

Reply:
What is the `gtIndexBaseType` actually needed for? Can it not be inferred from the rest of the signature?